lot/views.py

```diff
@@ -22,3 +27,37 @@ class LotListView(viewsets.ModelViewSet):
     ordering = ['-closing_date']
     filter_backends = [DjangoFilterBackend, AuctionTypeFilter]
     filterset_fields = ['auction__auction_status', ]
+
+    @action(detail=True)
+    def make_offer(self, request, pk=None):
+        lot = self.get_object()
+        serializer = OfferSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            validate_type_auction_english(lot)
+            validate_status(lot)
+            offer_price = serializer.validated_data['price']
+            validate_offer_price(lot, offer_price)
+
+            Offer.objects.create(user=request.user, lot=lot, price=offer_price)
+
+            lot.auction.current_price = offer_price
+            lot.auction.save(update_fields=['current_price'])
+
+            return Response({"message": "Your offer has been accepted."}, status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Приучай себя избавляться от лишней вложенности
```suggestion
        if not serializer.is_valid(raise_exception=True):
            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
           
        validate_type_auction_english(lot)
        validate_status(lot)
        offer_price = serializer.validated_data['price']
        validate_offer_price(lot, offer_price)

        Offer.objects.create(user=request.user, lot=lot, price=offer_price)

        lot.auction.current_price = offer_price
        lot.auction.save(update_fields=['current_price'])

        return Response({"message": "Your offer has been accepted."}, status=status.HTTP_201_CREATED)
```